src/app.js

```diff
@@ -0,0 +1,247 @@
+                    singleTransform: {
+                        type: "fluid.transforms.free",
+                        func: "gpii.taskTray.menu.updateMenu",
+                        args: ["{that}", "{app}.model.gpiiStarted", "{app}.model.keyedInSet"]
+                    }
+                },
+                invokers: {
+                    // The function used to apply prefs sets
+                    changeSet: {
+                        changePath: "{app}.model.keyedInSet",
+                        value: "{arguments}.0"
+                    }
+                }
+            }
+        }
+    },
+    events: {
+        onAppReady: null
+    },
+    listeners: {
+        "onCreate.updateTaskTray": {
+            funcName: "{that}.updateTaskTray",
+            args: ["{that}.model.menu"]
+        }
+    },
+    modelListeners: {
+        "menu": {
+            funcName: "{that}.updateTaskTray",
+            args: ["{change}.value"],
+            excludeSource: "init"
+        }
+    },
+    invokers: {
+        updateTaskTray: {
+            funcName: "gpii.taskTray.updateTaskTray",
+            args: ["{that}.tray", "{that}.options.labels.tooltip", "{arguments}.0"] // menu
+        }
+    },
+    icon: "icons/gpii.ico",
+    labels: {
+        tooltip: "GPII Electron"
+    }
+});
+
+gpii.taskTray.makeTray = function (icon) {
+    return new Tray(path.join(__dirname, icon));
+};
+
+gpii.taskTray.updateTaskTray = function (tray, tooltipLabel, menu) {
+    tray.setToolTip(tooltipLabel);
+    tray.setContextMenu(Menu.buildFromTemplate(menu));
+};
+
+/*
+ ** Component to start and stop parts of the GPII as well as key in and key out users
+ */
+fluid.defaults("gpii.app", {
+    gradeNames: "fluid.modelComponent",
+    model: {
+        gpiiStarted: false,
+        keyedInSet: null
+    },
+    events: {
+        onAppReady: null
+    },
+    listeners: {
+        "onCreate.startGpii": {
+            funcName: "gpii.app.startLocalFlowManager",
+            args: ["{that}"]
+        },
+        "onCreate.changeStarted": {
+            funcName: "{that}.changeStarted",
+            args: [true]
+        }
+    },
+    modelListeners: {
+        keyedInSet: [{
+            funcName: "gpii.app.keyOut",
+            args: ["{change}.oldValue"]
+        }, {
+            funcName: "gpii.app.keyIn",
+            args: ["{change}.value"],
+            priority: "after:keyOut"
+        }]
+    },
+    invokers: {
+        changeStarted: {
+            changePath: "gpiiStarted",
+            value: "{arguments}.0"
+        }
+    }
+});
+
+gpii.app.startLocalFlowManager = function (that) {
+    var fluid = require("universal"),
+        gpii = fluid.registerNamespace("gpii");
+
+    if (os.platform() === "win32") {
+        require("gpii-windows/index.js");
+    }
+
+    gpii.start();
+    //TODO: needs to fire an on ready so we can be certain that everything has started
+    that.events.onAppReady.fire();
```

Comment:
It is a little awkward, but there is a certain entitlement that an express server may actually start asynchronously (although I have not yet observed this in practice). To be absolutely secure that the GPII has really started, you will need to inject down a listener to this event in the Kettle server: https://github.com/fluid-project/kettle/blob/master/lib/KettleServer.js#L68
If we are short of time this can be left as a TODO since it will require a sound understanding of the geometry of everything as well as a bit of options distribution 
http://docs.fluidproject.org/infusion/development/IoCSS.html#-distributeoptions-and-iocss-downward-matching-css-like-context-selectors-for-options-forwarding